pkg/prometheus/host_filter.go

```diff
@@ -154,9 +161,18 @@ func shouldFilterTarget(target model.LabelSet, common model.LabelSet, host strin
 		return false
 	}
 
-	// Fall back to testing __meta_kubernetes_pod_node_name
-	if hostAddress := lset.Get(kubernetesNodeNameLabel); hostAddress != "" {
-		return shouldFilterTargetByLabelValue(hostAddress)
+	// Fall back to checking metalabels as long as their values are nonempty
+	for _, check := range labelMatchers {
+		addr := lset.Get(check)
```

Comment:
nit: I think you can inline this check?

Reply:
yeah, my initial attempt was

```go
if addr := lset.Get(check); addr == "" && !shouldFilterTargetByLabelValue(addr) {
  return false
}
```

but I thought that was a little hard to read. WDYT?

Reply:
Good thinking, I like the way it is now. Just leave it as is.